src/panels/developer-tools/state/developer-tools-state.js

```diff
@@ -412,72 +411,63 @@ class HaPanelDevState extends EventsMixin(LocalizeMixin(PolymerElement)) {
   }
 
   computeEntities(hass, _entityFilter, _stateFilter, _attributeFilter) {
-    const _entityFilterLength = _entityFilter && _entityFilter.length;
-    const _entityFilterLow = _entityFilter && _entityFilter.toLowerCase();
+    const entityFilterRegExp =
+      _entityFilter && RegExp(_entityFilter.replace(/\*/g, ".*"), "i");
```

Comment:
What about the `.` people might enter?

Reply:
Mmmm we should escape that you think?

Reply:
Let's use this method: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Regular_Expressions#escaping

And then find the `\*` and replace that with `.*`